File: dotnet/Lexy.Poc/Lexy.Poc.Core/Parser/Tokens/MemberAccessLiteral.cs
using System;

namespace Lexy.Poc.Core.Parser.Tokens
{
    public class MemberAccessLiteral : Token, ILiteralToken
    {
        public override string Value { get; }

        public MemberAccessLiteral(string value)
        {
            Value = value ?? throw new ArgumentNullException(nameof(value));
        }

        public string[] GetParts() => Value.Split(TokenValues.MemberAccess);

        public override string ToString() => Value;
    }
}

File: typescipt/src/parser/tokens/memberAccessLiteral.ts


export class MemberAccessLiteral extends Token, ILiteralToken {
   public string Parent => Parts.Length >= 1 ? Parts[0] : null;
   public string Member => Parts.Length >= 2 ? Parts[1] : null;

   public string[] Parts

   public MemberAccessLiteral(string value, TokenCharacter character) : base(character) {
     Value = value ?? throw new Error(nameof(value));
     Parts = Value.Split(TokenValues.MemberAccess);
   }

   public override string Value

   public object TypedValue => Parts;

   public deriveType(context: IValidationContext): VariableType {
     let variableReference = new VariableReference(Parts);
     let variableType = context.VariableContext.GetVariableType(variableReference, context);
     if (variableType != null) return variableType;

     if (Parts.Length != 2) return null;

     let rootType = context.RootNodes.GetType(Parent);
     return rootType is not ITypeWithMembers typeWithMembers ? null : typeWithMembers.MemberType(Member, context);
   }

   public override toString(): string {
     return Value;
   }
}
